com.ibm.wala.core/src/com/ibm/wala/classLoader/ClassLoaderImpl.java

```diff
@@ -687,19 +673,11 @@ public Reader getSource(IClass klass) {
   /*
    * @see com.ibm.wala.classLoader.IClassLoader#removeAll(java.util.Collection)
    */
-  @SuppressWarnings("unused")
   @Override
   public void removeAll(Collection<IClass> toRemove) {
-    if (toRemove == null) {
-      throw new IllegalArgumentException("toRemove is null");
-    }
-    for (IClass klass : toRemove) {
-      if (DEBUG_LEVEL > 0) {
-        System.err.println("removing " + klass.getName());
-      }
-      loadedClasses.remove(klass.getName());
-      sourceMap.remove(klass.getName());
-    }
+    toRemove.stream().map(IClass::getName)
```

Comment:
Retain the null check on the argument

Reply:
Oops, good catch. I restore the `null` check this in a forthcoming group of updates for this pull request. Thank you for reviewing these proposed changes so attentively!